core/src/main/java/com/orientechnologies/orient/core/sql/executor/OSelectExecutionPlanner.java

```diff
@@ -113,6 +73,10 @@ private void init() {
     info.limit = this.statement.getLimit();
     info.lockRecord = this.statement.getLockRecord();
     info.timeout = this.statement.getTimeout() == null ? null : this.statement.getTimeout().copy();
+    if (info.timeout == null && OGlobalConfiguration.COMMAND_TIMEOUT.getValueAsLong() > 0) {
+      info.timeout = new OTimeout(-1);
+      info.timeout.setVal(OGlobalConfiguration.COMMAND_TIMEOUT.getValueAsLong());
```

Comment:
should use the context configuration here not the static value as well

Reply:
Done